modules/lazyload/module.php

```diff
@@ -38,38 +38,118 @@ private function enqueue_styles() {
 		);
 	}
 
+	private function get_repeaters_selectors( $controls, $settings ) {
+		$selectors = [];
+		$repeater = array_filter( $controls, function( $control ) {
+			if ( 'repeater' === $control['type'] && Utils::get_array_value_by_keys( $control, [ 'fields', 'background_image', 'fields_options', 'image', 'background_lazyload', 'active' ] ) ) {
+				return true;
+			}
+		} );
+		if ( empty( $repeater ) ) {
+			return false;
+		}
+		$repeater = array_shift( $repeater );
+
+		if ( $repeater ) {
+			$repeater_settings = $settings[ $repeater['name'] ][0];
+			$repeater_selector = $this->get_repeater_selector( $repeater_settings, $repeater );
+			if ( $repeater_selector ) {
+				$selectors[] = $repeater_selector;
+			}
+		}
+
+		return $selectors;
+	}
+
+	private function get_repeater_selector( $repeater_settings, $repeater ) {
+		$lazyload_options = Utils::get_array_value_by_keys( $repeater, [ 'fields', 'background_image', 'fields_options', 'image', 'background_lazyload' ] );
+		if ( ! $lazyload_options['active'] ) {
+			return false;
+		}
+
+		$repeater_selector = Utils::get_array_value_by_keys( $lazyload_options, [ 'selector' ] );
+		if ( ! $repeater_selector ) {
+			return false;
+		}
+
+		$repeater_keys = Utils::get_array_value_by_keys( $repeater_settings, [ 'background_lazyload', 'keys' ] );
+		$repeater_background_image_url = Utils::get_array_value_by_keys( $repeater_settings, $repeater_keys );
+
+		if ( $repeater_background_image_url ) {
+			return $repeater_selector;
+		}
+
+		return false;
+	}
+
 	private function update_element_attributes( Element_Base $element ) {
+
 		$settings = $element->get_settings_for_display();
 		$controls = $element->get_controls();
 		$lazyload_attribute_name = 'data-e-bg-lazyload';
-
+		$attributes = [];
 		$controls_with_background_image = array_filter( $controls, function( $control ) {
 			return Utils::get_array_value_by_keys( $control, [ 'background_lazyload', 'active' ] );
 		} );
 
+		$repeaters_selectors = $this->get_repeaters_selectors( $controls, $settings );
+
 		foreach ( $controls_with_background_image as $control_name => $control_data ) {
+
+			$selectors = [];
+
+			if ( $repeaters_selectors ) {
+				$selectors = array_merge( $repeaters_selectors, $selectors );
+			}
```

Comment:
Why it is inside the foreach loop

Reply:
It is inside the loop because of it's running for each control inside the element. For each control has his own `$selectors` & `$repeaters_selectors`. If I move it outside the for each loop, the lazyload will not work. Did i miss anything?